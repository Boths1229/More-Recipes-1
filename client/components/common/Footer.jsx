import React from 'react';

const Footer = () => (
  <footer className="footer-section" id="footer-below">
    <div className="container">
      <strong
        className="copyright"
      >
        More-Recipes Â© 2018, All Rights Reserved, Design &amp; Developed By: SannyTee
      </strong>
    </div>
  </footer>
);

export default Footer;
