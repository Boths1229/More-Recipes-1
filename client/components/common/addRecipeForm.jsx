import React, { PropTypes } from 'react';
import { Pulse } from 'react-preloading-component';
import firebase from 'firebase';
import ImageUploader from 'react-firebase-image-uploader';

const propTypes = {
  errorMessage: PropTypes.string,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  isUploading: PropTypes.bool.isRequired,
  progress: PropTypes.number.isRequired,
  handleProgress: PropTypes.func.isRequired,
  handleUploadStart: PropTypes.func.isRequired,
  handleUploadSuccess: PropTypes.func.isRequired,
  mealType: PropTypes.string.isRequired,
};

const defaultProps = {
  errorMessage: '',
};

const AddRecipeForm = props => (
  <div
    className="modal fade"
    id="exampleModal"
    tabIndex="-1"
    data-backdrop="static"
    role="dialog"
    aria-labelledby="exampleModalLabel"
    style={{ display: 'none' }}
    aria-hidden="true"
  >
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header black">
          <h5 className="modal-title black" id="exampleModalLabel">Add new recipe</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">Ã—</span>
          </button>
        </div>
        <div className="modal-body">
          {
            props.errorMessage &&
            <div className="alert alert-danger">
              {
                props.errorMessage
              }
            </div>
          }

          <form onSubmit={props.handleSubmit} id="recipeForm">
            <div className="form-group">
              <label
                htmlFor="recipient-name"
                className="col-form-label black"
              >
                Title:
              </label>
              <input
                type="text"
                name="recipeName"
                className="form-control"
                id="recipient-name"
                required
                onChange={props.handleChange}
                onFocus={props.onFocus}
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="message-text"
                className="col-form-label black"
              >
                Description:
              </label>
              <textarea
                className="form-control"
                id="message-text"
                name="description"
                required
                onChange={props.handleChange}
                onFocus={props.onFocus}
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="image"
                className="col-form-label black"
              >
                Image:
              </label>
              {
                props.isUploading &&
                props.progress < 100 &&
                <div>
                  <Pulse />
                </div>
              }
              <ImageUploader
                id="uploadImage"
                name="image"
                storageRef={
                  firebase
                    .storage()
                    .ref('images')
                }
                onProgress={props.handleProgress}
                onUploadSuccess={props.handleUploadSuccess}
                onUploadStart={props.handleUploadStart}
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="type-text"
                className="col-form-label black"
              >
                Type of meal:
              </label>
              <select
                value={props.mealType}
                name="mealType"
                className="form-control"
                id="type-text"
                required
                onChange={props.handleChange}
                onFocus={props.onFocus}
              >
                <option defaultValue="breakfast">Breakfast</option>
                <option value="brunch">Brunch</option>
                <option value="elevenses">Elevenses</option>
                <option value="lunch">Lunch</option>
                <option value="tea">Tea</option>
                <option value="supper">Supper</option>
                <option value="dinner">Dinner</option>
              </select>
            </div>
            <div className="form-group">
              <label
                htmlFor="ingredients-text"
                className="col-form-label black"
              >
                Ingredients:
              </label>
              <textarea
                placeholder="separate each ingredient with a comma"
                name="ingredients"
                className="form-control"
                id="ingredients"
                required
                onChange={props.handleChange}
                onFocus={props.onFocus}
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="method-text"
                className="col-form-label black"
              >
                Method of cooking:
              </label>
              <textarea
                name="method"
                required
                onChange={props.handleChange}
                onFocus={props.onFocus}
                className="form-control"
                id="method-text"
              />
            </div>
            <div className="modal-footer">
              <button
                type="button"
                className="btn btn-secondary"
                data-dismiss="modal"
              >
                Close
              </button>
              <button
                type="submit"
                className="btn vote-button"
              >
                Add recipe
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
);

AddRecipeForm.propTypes = propTypes;
AddRecipeForm.defaultProps = defaultProps;

export default AddRecipeForm;
