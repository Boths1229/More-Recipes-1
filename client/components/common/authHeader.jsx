/* eslint-disable no-unused-vars */
/**
 *  @fileOverview Creates header for authenticated users
 *
 *  @author       Sanni Taiwo
 *
 *  @requires     NPM:react
 *  @requires     NPM:react-router
 *  @requires     NPM:react-redux
 *  @requires     NPM:redux
 *  @requires     NPM:toastr
 */
import React, { Component } from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import { createRecipeAction } from '../../actions/recipesAction';


/**
 * @description class to represent authenticated user header
 * @extends Component
 */
class authHeader extends Component {
  /**
   * @description Creates a recipe
   * @param {object} props
  */
  constructor(props) {
    super(props);
    this.state = {
      recipeName: '',
      mealType: '',
      description: '',
      method: '',
      ingredients: '',
      errorMessage: '',
      errors: []
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onFocus = this.onFocus.bind(this);
  }

  /**
   * @description checks for update in form entry
   * @param {object} event
   * @memberof authHeader
   * @returns {void}
  */
  handleChange(event) {
    const { name } = event.target,
      { value } = event.target;
    this.setState({
      [name]: value
    });
  }

  /**
   * @description checks when an element is being focused on
   * @param {object} event
   * @memberof authHeader
   * @returns {void}
  */
  onFocus(event) {
    this.setState({ errorMessage: '' });
  }

  /**
   * @description handles form submission
   * @param {object} event
   * @memberof authHeader
   * @returns {void}
  */
  handleSubmit(event) {
    event.preventDefault();
    this.props.actions.createRecipeAction(this.state)
      .then(() => {
        toastr.success('Recipe successfully added to catalog');
        document.getElementById('recipeForm').reset();
        $('#exampleModal').modal('hide');
      })
      .catch((err) => {
        const { data } = err.response;
        if (data.error) {
          this.setState({
            errorMessage: 'Pleae select a meal type'
          });
        } else {
          data.map((message) => {
            this.setState({
              errorMessage: message
            });
          });
        }
      });
  }


  /**
   * @description renders the component
   * @memberof authHeader
   * @returns {void} returns the navbar section of authenticated users
  */
  render() {
    return (
      <header>
        <nav className="navbar fixed-top  navbar-expand-lg navbar-dark" style={{ backgroundColor: '#2b3034' }}>
          <a className="navbar-brand" href="/">More-Recipes</a>
          <button className="navbar-toggler"
            type="button"
            data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation">
              <span className="navbar-toggler-icon"></span>
          </button>

          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <form className="form-inline my-2 my-lg-0 ml-auto ">
              <input
                className="form-control mr-sm-2"
                type="search"
                placeholder="Search"
                aria-label="Search" />
            </form>
            <ul className="navbar-nav ml-auto">
              <li className="nav-item">
                <button type="button"
                  className="btn vote-button rfloat"
                  style= {{ float: 'right' }}
                  data-toggle="modal"
                  data-target="#exampleModal">
                    <i className="fa fa-plus-square-o" aria-hidden="true"></i>
                </button>
              </li>
              <li className="nav-item">
                <span className="nav-link" >Logged in as {this.props.user.username}</span>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="profile">Profile</Link>
              </li>
              <li className="nav-item">
                <Link className="nav-link" to="/my-recipes">My Recipes</Link>
              </li>
              <li className="nav-item">
                <a className="nav-link disabled">Signout</a>
              </li>
            </ul>
          </div>
        </nav>
        <div>
          <div
            className="modal fade"
            id="exampleModal"
            tabindex="-1"
            data-backdrop="static"
            role="dialog"
            aria-labelledby="exampleModalLabel"
            style={{ display: 'none' }}
            aria-hidden="true">
              <div className="modal-dialog" role="document">
                <div className="modal-content">
                  <div className="modal-header black">
                    <h5 className="modal-title black" id="exampleModalLabel">Add new recipe</h5>
                    <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                      <span aria-hidden="true">Ã—</span>
                    </button>
                  </div>
                  <div className="modal-body">
                  {
                    this.state.errorMessage &&
                    <div className="alert alert-danger">
                      {
                        this.state.errorMessage
                      }
                    </div>
                  }
                  {

                  }
                    <form onSubmit={this.handleSubmit} id="recipeForm">
                      <div className="form-group">
                        <label for="recipient-name" className="col-form-label black">Title:</label>
                        <input
                          type="text"
                          name="recipeName"
                          className="form-control"
                          id="recipient-name"
                          required
                          onChange={this.handleChange}
                          onFocus={this.onFocus}
                        />
                      </div>
                      <div className="form-group">
                        <label
                          for="message-text"
                          className="col-form-label black">
                          Description:
                            </label>
                        <textarea
                          className="form-control"
                          id="message-text"
                          name="description"
                          required
                          onChange={this.handleChange}
                          onFocus={this.onFocus}>
                        </textarea>
                      </div>
                      <div className="form-group">
                        <label
                          for="type-text"
                          className="col-form-label black">
                          Type of meal:
                            </label>
                        <select
                          value={this.state.mealType}
                          name="mealType"
                          className="form-control"
                          id="type-text"
                          required
                          onChange={this.handleChange}
                          onFocus={this.onFocus}>
                            <option value="breakfast">Breakfast</option>
                            <option value="brunch">Brunch</option>
                            <option value="elevenses">Elevenses</option>
                            <option value="lunch">Lunch</option>
                            <option value="tea">Tea</option>
                            <option value="supper">Supper</option>
                            <option value="dinner">Dinner</option>
                        </select>
                      </div>
                      <div className="form-group">
                        <label
                          for="ingredients-text"
                          className="col-form-label black">
                          ingredients:
                            </label>
                        <textarea
                          placeholder="separate each ingredient with a comma"
                          name="ingredients"
                          className="form-control"
                          id="ingredients"
                          required
                          onChange={this.handleChange}
                          onFocus={this.onFocus}>
                        </textarea>
                      </div>
                      <div className="form-group">
                        <label
                          for="method-text"
                          className="col-form-label black">
                          Method of cooking:
                            </label>
                        <textarea
                          name="method"
                          required
                          onChange={this.handleChange}
                          onFocus={this.onFocus}
                          className="form-control"
                          id="method-text">
                        </textarea>
                      </div>
                      <div className="modal-footer">
                        <button
                          type="button"
                          className="btn btn-secondary"
                          data-dismiss="modal">
                            Close
                          </button>
                        <button type="submit" className="btn vote-button">Add recipe</button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
          </div>
        </div>
      </header>
    );
  }
}

/**
 * @description maps action to properties of authHeader
 * @param  {object} dispatch
 * @returns {object} returns the action to be bind
 */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({
      createRecipeAction,
    }, dispatch)
  };
}

export default connect(null, mapDispatchToProps)(authHeader);
