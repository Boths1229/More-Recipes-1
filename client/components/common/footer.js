import React from 'react';

/* eslint-disable no-unused-vars */

const Footer = (props) => {
  return (
    <footer className="footer-section" id="footer-below">
      <div className="container">
        <strong className="copyright">More-Recipes Â© 2017, All Rights Reserved, Design &amp; Developed By: SannyTee</strong>
      </div>
    </footer>
  );
}

export default Footer;