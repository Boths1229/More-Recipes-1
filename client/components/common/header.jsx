/* eslint-disable no-unused-vars */
import React from 'react';
import { Link } from 'react-router';


const Header = (props) => {
  return (
    <header>
      <nav className="navbar navbar-expand-lg navbar-dark" style={{ backgroundColor: '#2b3034' }}>
        <a className="navbar-brand" href="#">More-Recipes</a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <form className="form-inline my-2 my-lg-0 ml-auto ">
            <input
              className="form-control mr-sm-2"
              type="search"
              placeholder="Search"
              aria-label="Search"/>
          </form>
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link className="nav-link" to="/profile">Profile</Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/users/:userId/recipe">My Recipes</Link>
            </li>
            <li className="nav-item">
              <a className="nav-link disabled" href="#">Signout</a>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  );
};

export default Header;
