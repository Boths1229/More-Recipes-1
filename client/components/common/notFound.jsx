import React from 'react';
import { Link } from 'react-router';
import Footer from './Footer';


const NotFound = () => (
  <div>
    <header>
      <nav className="navbar fixed-top navbar-expand-lg navbar-dark" style={{ backgroundColor: '#2b3034' }}>
        <a className="navbar-brand" href="/">More-Recipes</a>
        <button
          className="navbar-toggler"
          type="button"
          data-toggle="collapse"
          data-target="#navbarSupportedContent"
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>

        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav ml-auto">
            <li className="nav-item">
              <Link className="nav-link navbar-brand" to="/recipes">Home</Link>
            </li>
          </ul>
        </div>
      </nav>
    </header>
    <div className="container main">
      <div className="pt-5 main-div" style={{ textAlign: 'center' }}>
        <img
          src="http://clipartmag.com/images/oops-emoticon-17.jpg"
          name="image"
          alt="error"
          className="img-fluid rounded-circle"
        />
        <h1>Page Not Found</h1>
        <div >
          <Link id="remove-link" className="vote-button errorBtn" to="/recipes">Go Back</Link>
        </div>
      </div>
    </div>
    <Footer />
  </div>
);

export default NotFound;
