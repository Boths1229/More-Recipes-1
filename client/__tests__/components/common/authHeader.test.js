import React from 'react';
import { shallow } from 'enzyme';
import { AuthHeader } from '../../../components/common/AuthHeader';
import MockData from '../../__mocks__/actions/recipes';


let props;

const setup = () => {
  props = {
    actions: {
      createRecipeAction: () => Promise.resolve(),
      logoutAction: () => {},
    },
    user: MockData.userData.profile
  };

  return shallow(<AuthHeader {...props} />);
};

describe('AuthHeader Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('header').length).toBe(1);
    expect(wrapper.find('nav').length).toBe(1);
    expect(wrapper.find('button').length).toBe(2);
    expect(wrapper.find('ul').length).toBe(1);
  });

  it('should set state of isUploading and progress when image upload starts', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const startUpload = jest.spyOn(action, 'handleUploadStart');
    action.handleUploadStart();
    expect(startUpload).toBeCalled();
    expect(wrapper.state('isUploading')).toBe(true);
    expect(wrapper.state('progress')).toBe(0);
  });

  it('should track state of progress when image is uploading', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const trackProgress = jest.spyOn(action, 'handleProgress');
    action.handleProgress(20);
    expect(trackProgress).toBeCalled();
    expect(wrapper.state('progress')).toBe(20);
  });

  it('should set state of input fields when input value changes', () => {
    const event = {
      target: { name: 'recipeName', value: 'Rice' }
    };
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'handleChange');
    action.handleChange(event);
    expect(spy).toBeCalled();
    expect(wrapper.state('recipeName')).toBe('Rice');
  });

  it('should set state of backspaceRemoves when `toggleBackspaceRemoves` method is called', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const toggleBackspace = jest.spyOn(action, 'toggleBackspaceRemoves');
    action.toggleBackspaceRemoves();
    expect(toggleBackspace).toBeCalled();
    expect(wrapper.state('backspaceRemoves')).toBe(false);
  });

  it('should clear error when input field is focused on', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    wrapper.setState({ errorMessage: 'Upload an image' });
    const clearError = jest.spyOn(action, 'onFocus');
    action.onFocus({ preventDefault: () => {} });
    expect(clearError).toBeCalled();
    expect(wrapper.state('errorMessage')).toBe('');
  });

  it('should set state of value when `onSearchChange` method is called', () => {
    const wrapper = setup();
    const action = wrapper.instance();
    const search = jest.spyOn(action, 'onSearchChange');
    action.onSearchChange('recipe');
    expect(search).toBeCalled();
    expect(wrapper.state('value')).toBe('recipe');
  });

  it('should submit form to add recipe', () => {
    document.getElementById = () => ({ reset: jest.fn() });
    const wrapper = setup();
    const action = wrapper.instance();
    const spy = jest.spyOn(action, 'handleSubmit');
    action.handleSubmit({ preventDefault: () => {} });
    expect(spy).toBeCalled();
  });
});
