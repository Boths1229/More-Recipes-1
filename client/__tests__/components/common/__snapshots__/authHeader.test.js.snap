// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader Component should render correctly 1`] = `
<header>
  <nav
    className="navbar fixed-top  navbar-expand-lg navbar-dark"
    style={
      Object {
        "backgroundColor": "#2b3034",
      }
    }
  >
    <Link
      className="navbar-brand"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/recipes"
    >
      More-Recipes
    </Link>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <div
        className="form-inline my-2 my-lg-0 ml-auto "
      >
        <div
          className="section"
          style={
            Object {
              "width": "300px",
            }
          }
        >
          <Async
            autoload={true}
            backspaceRemoves={true}
            cache={Object {}}
            filterOptions={false}
            ignoreAccents={true}
            ignoreCase={true}
            labelKey="recipeName"
            loadOptions={[Function]}
            loadingPlaceholder="Loading..."
            multi={false}
            noResultsText="No recipe found"
            onChange={[Function]}
            onValueClick={[Function]}
            options={Array []}
            placeholder="Search for recipe"
            searchPromptText="Type to search for recipe"
            value=""
            valueKey="id"
          />
        </div>
      </div>
      <ul
        className="navbar-nav ml-auto"
      >
        <li
          className="nav-item"
        >
          <button
            className="btn vote-button rfloat"
            data-target="#exampleModal"
            data-toggle="modal"
            style={
              Object {
                "float": "right",
              }
            }
            type="button"
          >
            Add Recipe
          </button>
        </li>
        <li
          className="nav-item"
        >
          <span
            className="nav-link"
          >
            Logged in as 
            sannytee
          </span>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/my-profile"
          >
            Profile
          </Link>
        </li>
        <li
          className="nav-item dropdown show"
        >
          <Link
            aria-expanded="true"
            aria-haspopup="true"
            className="nav-link dropdown-toggle"
            data-toggle="dropdown"
            id="navlink"
            onlyActiveOnIndex={false}
            role="button"
            style={Object {}}
            to="#"
          >
            Recipes
          </Link>
          <div
            aria-labelledby="navlink"
            className="dropdown-menu"
          >
            <Link
              className="dropdown-item"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/my-recipes"
            >
              My Recipes
            </Link>
            <Link
              className="dropdown-item nav-text"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/my-favorite"
            >
              Favorite recipes
            </Link>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            id="logout"
            onClick={[Function]}
            role="none"
          >
            Signout
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <AddRecipeForm
    errorMessage=""
    handleChange={[Function]}
    handleProgress={[Function]}
    handleSubmit={[Function]}
    handleUploadStart={[Function]}
    handleUploadSuccess={[Function]}
    isUploading={false}
    mealType="breakfast"
    notReady={true}
    onFocus={[Function]}
    progress={0}
  />
</header>
`;
